packages/dataviews-client/src/resolve-dataviews.ts

```diff
@@ -68,39 +68,141 @@ export const mergeWorkspaceUrlDataviewInstances = (
   return [...urlDataviews.new, ...workspaceDataviewInstancesMerged]
 }
 
-export const resolveDataviewDatasetResource = (
+const getDatasetConfigsByDatasetType = (
+  dataview: UrlDataviewInstance,
+  type: DatasetTypes
+): DataviewDatasetConfig[] => {
+  const availableDatasets = dataview.datasets || []
+  const availableDatasetConfigs = dataview.datasetsConfig || []
+  const datasets = availableDatasets.filter((dataset) => dataset.type === type)
+  const datasetIds = datasets.map((dataset) => dataset.id)
+  const datasetConfigs = availableDatasetConfigs.filter((datasetConfig) =>
+    datasetIds.includes(datasetConfig.datasetId)
+  )
+  return datasetConfigs
+}
+
+const getDatasetConfigByDatasetType = (
   dataview: UrlDataviewInstance,
-  typeOrId: DatasetTypes | DatasetTypes[] | string
-): {
-  dataset?: Dataset
-  datasetConfig?: DataviewDatasetConfig
-  url?: string
-} => {
-  const isArray = Array.isArray(typeOrId)
-  const isType = isArray || Object.values(DatasetTypes).includes(typeOrId as DatasetTypes)
+  type: DatasetTypes
+): DataviewDatasetConfig => {
+  return getDatasetConfigsByDatasetType(dataview, type)[0]
+}
+
+export type TrackDatasetConfigs = {
+  info: DataviewDatasetConfig
+  track: DataviewDatasetConfig
+  events: DataviewDatasetConfig[]
+}
+
+const getTrackDataviewDatasetConfigs = (
+  dataviewInstance: UrlDataviewInstance
+): TrackDatasetConfigs => {
+  const info = getDatasetConfigByDatasetType(dataviewInstance, DatasetTypes.Vessels)
+
+  const trackDatasetType =
+    dataviewInstance.datasets && dataviewInstance.datasets?.[0]?.type === DatasetTypes.UserTracks
+      ? DatasetTypes.UserTracks
+      : DatasetTypes.Tracks
+  const track = { ...getDatasetConfigByDatasetType(dataviewInstance, trackDatasetType) }
+
+  const events = getDatasetConfigsByDatasetType(dataviewInstance, DatasetTypes.Events).filter(
+    (datasetConfig) => datasetConfig.query?.find((q) => q.id === 'vessels')?.value
+  ) // Loitering
+
+  return {
+    info,
+    track,
+    events,
+  }
+}
+
+export const getDataviewsForResourceQuerying = (
+  dataviewInstances: UrlDataviewInstance[],
+  trackDatasetConfigsTransform?: (
```

Comment:
This seems to be very adhoc for the track datasets... and the idea is to keep it as generic as possible. so why not having something like `DatasetConfigsTransform` being a dictionary of dataview.type as keys and the function as the value, this way we could use the function in any dataview type